tests/functional/api/groups/read_test.py

```diff
@@ -35,8 +38,8 @@ def test_it_overrides_authority_param_with_user_authority(
         self, app, factories, db_session, user_with_token, token_auth_header
     ):
         user, _ = user_with_token
-        # This group will be created with the user's authority
-        group1 = factories.Group(creator=user)
+        group1 = factories.Group(authority=user.authority)
+        db_session.add(GroupMembership(group_id=group1.id, user_id=user.id))
```

Comment:
Here the test was relying on the fact that `group1.authority` would be the same as `user.authority`, which has nothing to do with `user` being `group1`'s `creator` and is actually just a coincidence (because both `factories.User()` and `factories.Group()` default `authority` to `"example.com"`). I've fixed the test to specify the group's authority explicitly and to remove the irrelevant concept of "creator".